### PR TITLE
feat: add Idempotency-Key header and exponential retry mechanism

### PR DESCRIPTION
**Pull request by Joao-Moura**
- Add new header `Idempotency-Key` using UUID v4
- Expands exponential retry mechanism from the req library
- Add new environments to configure retry
- Update req lib (to have the retry step) and add elixir_uuid (to have UUID generation)

### CODE REVIEW

File: lib/novu/http.ex [hunk 1]
```diff
@@ -91,7 +110,8 @@ defmodule Novu.Http do
     [
       {"accept", "application/json"},
       {"content-type", "application/json"},
-      {"authorization", "ApiKey #{api_key()}"}
+      {"authorization", "ApiKey #{api_key()}"},
+      {"Idempotency-Key", UUID.uuid4()}
```
Comment (doomspork) [inline lib/novu/http.ex:114]: Do we need the case sensitivity here?

```suggestion
      {"idempotency-key", UUID.uuid4()}
```

**Review by doomspork — COMMENTED** [review]
Thank you @Joao-Moura! One quick question on this but otherwise this is looking good. I'll pull this down and test before approving.

File: lib/novu/http.ex (hunk 1 above)
Comment (Joao-Moura) [reply lib/novu/http.ex:114]: No, it's unnecessary

**Review by btkostner — APPROVED** [review]
Code looks good to me

**Pull request merged**